### ISSUE TITLE
Estimate effort to support pull request panel operations with BitBucket (public and private)

### ISSUE THREAD

**Issue by bmicklea** [eclipse-che/che#3052]
Today our pull request panel operations only work with a public GitHub repo. The same operations should be possible with a BitBucket or BitBucket Server (previously known as Stash) repo.

This task is to estimate the number of sprints needed to implement all the same pull request operations with BitBucket and BitBucket Server (public and private) repos.

**Comment by vinokurig** [comment]
> Today our pull request panel operations only work with a public GitHub repo.

Actually it works also with BitBucket and Microsoft VSTS.

**Comment by bmicklea** [comment]
@vinokurig - Thanks for the clarification but I don't think this works with BitBucket Server (on-premises private install of BitBucket). Can you please test that.

**Comment by vinokurig** [comment]
@bmicklea ok, I will check, when BitBucket Server instance will be ready. @dzamosenchuk should prepare it.

**Comment by vinokurig** [comment]
@bmicklea I have checked and you was right, it doesn't work with BitBucket Server. I am fixing it now.

**Comment by bmicklea** [comment]
Great thanks Igor

**Comment by vinokurig** [comment]
@bmicklea I have discovered that our pull request plugin can work only with hosted BitBucket because the URL of BitBucket API in our service is hard-coded to `bitbucket.org`, that is different from on-premises BitBucket end point. The same is for GitHub and Microsoft VSTS. I think we need separate issue for implementing multiple end point to BitBucket, GitHub and Microsoft VSTS services, to be able to work with their on-premises versions.

**Comment by bmicklea** [comment]
Why do we need a separate issue - we have two customers waiting on this.
It's urgent.

Can we do a simple fix that puts the URL in a configuration file? It
doesn't need to be a pretty solution but we need to get this done ASAP.

**Comment by vinokurig** [comment]
@bmicklea Sorry for not full information but also the API of BitBucket Server is different. Looks like we need to implement separate service for BitBucket Server.

**Comment by bmicklea** [comment]
Okay, I can create a separate issue but please start work on this now.

**Comment by bmicklea** [comment]
Closing this issue - we need to start work on this issue though as this support is critical: https://github.com/codenvy/enterprise/issues/27
